src/charm.py

```diff
@@ -240,6 +257,56 @@ def _on_get_admin_secrets_action(self, event: ActionEvent):
 
         event.set_results({"password": password if password else "", "chain": chain})
 
+    def _on_start_service_action(self, event: ActionEvent):
+        """Start the OpenSearch service from an action event."""
+        if self.opensearch.is_node_up():
+            event.set_results({"message": "OpenSearch is already started in this node."})
+            return
+
+        if not self._start_opensearch():
+            event.defer()
+            event.set_results(
+                {"message": "Something happened, the OpenSearch service will re-attempt a start."}
+            )
+            return
+
+        event.set_results({"message": "The OpenSearch service is attempting a start..."})
+
+    def _on_restart_service_action(self, event: ActionEvent):
+        """Restart the OpenSearch service from an action event."""
+        try:
+            self.unit.status = BlockedStatus(ServiceIsStopping)
+            self.opensearch.stop()
+            self.unit.status = BlockedStatus(ServiceStopped)
+
+            if not self._start_opensearch():
+                event.defer()
+                event.set_results(
+                    {
+                        "message": "Something happened, the OpenSearch service will re-attempt a start."
+                    }
+                )
+                return
+
+            event.set_results({"message": "The OpenSearch service is attempting a restart..."})
+        except OpenSearchStopError as e:
+            logger.error(e)
+            event.set_results(
+                {"message": "An error occurred during the stop of the OpenSearch service."}
+            )
+
+    def _on_stop_service_action(self, event: ActionEvent):
+        """Stop the OpenSearch service from an action event."""
+        try:
+            self.unit.status = BlockedStatus(ServiceIsStopping)
+            self.opensearch.stop()
+            self.unit.status = BlockedStatus(ServiceStopped)
+            event.set_results({"message": "The OpenSearch service is stopping..."})
+        except OpenSearchStopError:
+            event.set_results(
```

Comment:
See previous comment. Maybe is worth adding a blocked status as well, with a different message. Also add the error to the log.

Reply:
(For the comment regarding the use of BlockedStatus instead of Maintenance / Waiting etc.. This will be the part f another PR, as it requires to also change some of the integration tests etc.